pkg/build/controller/controller.go

```diff
@@ -317,7 +317,8 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 			}
 		}
 	case kapi.PodFailed:
-		build.Status.Reason = ""
+		// NOTE: why do we need to reset the reason for a failed Pod???
+		// build.Status.Reason = ""
```

Comment:
The "reason" is associated with why the build is in that phase.  So when you change the phase of the build, you need to clear the old reason because it's no longer applicable.